src/utils/internal.ts

```diff
@@ -847,3 +844,11 @@ export async function getCheckpointValue(
     ).schedule;
   }
 }
+
+export function isSameSet<T>(
```

Comment:
I think this should this be named `isSameArray` or `arrayDeepEqual`. 

imo Using the term Set is misleading with Array arguments since JS has an actual Set type.

Reply:
The thing is that order matters in arrays, so that name would be misleading. We could call it `isSameCollection`

Reply:
Good point. I like both names Jere suggested.